drafts/zip-daira-disable-sprout-outputs.rst

```diff
@@ -0,0 +1,156 @@
+
+For example, a vulnerability was discovered in the zero-knowledge proving system
+originally used by Zcash that could have allowed counterfeiting [#counterfeiting]_.
+While this particular vulnerability was addressed (also for Sprout shielded transactions)
+by the Sapling upgrade, and we are not aware of others at the time of writing, the
+possibility of other cryptographic weaknesses cannot be entirely ruled out.
+
+In addition, the Zcash specification and implementation incurs complexity and
+"technical debt" from the requirement to support both shielded transaction protocols.
+
+Removing the ability to send to Sprout outputs is a first step toward reducing this
+complexity and potential risk. This does not prevent extracting value held in Sprout
+addresses and sending it to transparent addresses, or to Sapling addresses via the
+migration tool [#migration]_.
+
+
+Specification
+=============
+
+A new v5 transaction format is defined, with the following changes relative to the
+Sapling v4 transaction format [#v4-tx]_:
+
+- In JoinSplit descriptions, the ``ephemeralKey``, and ``encCiphertexts`` fields are
+  removed.
+
+- When creating JoinSplit descriptions, the Sprout output notes MUST be either
+  internal change (i.e. spent in a subsequent JoinSplit of the same transaction),
+  or dummy notes as described in [#protocol]_ section 4.7.1.
+
+- The output commitments of a JoinSplit description in a v5 transaction MUST NOT
+  be added to the output Sprout commitment tree. That is, the root of the Sprout note
+  commitment tree in the output treestate of a v5 transaction is the same as that in
+  its input treestate. This does not affect the use of interstitial Sprout treestates
+  in v5 transactions.
+
+The removal of ``ephemeralKey`` and ``encCiphertexts`` fields means that for JoinSplit
+descriptions in v5 transactions, it is not necessary for a wallet to perform trial
+decryption of those fields to detect incoming payments as described in [#protocol]_
+sections 4.18 and 4.16.2. The processing of nullifiers in the algorithm of section 4.18
+is unaffected.
+
+Note that the consensus protocol does not prevent a wallet from incorrectly creating
+a Sprout output note that is neither internal change nor a dummy note. In this case
+the value associated with the note cannot be spent and is therefore "burnt".
+
+When this proposal is activated, nodes and wallets MUST disable any facilities to
+send to Sprout addresses, and this SHOULD be made clear in user interfaces.
+
+
+Rationale
+=========
+
+This design does not require any change to the JoinSplit circuit, thus minimizing
+the risk of security regressions, and avoiding the need for a new ceremony to generate
+circuit parameters.
+
+The code changes needed are relatively small and simple, and their security is easy
+to analyse.
+
+Removing the ``ephemeralKey`` and ``encCiphertexts`` fields reduces the size of a
```

Comment:
Further, it avoids bugs where wallets accept payments to their Sprout address without realizing the commitment won't end up actually being placed in the tree.

Reply:
No longer relevant.